compiler/rustc_typeck/src/check/generator_interior.rs

```diff
@@ -387,6 +387,18 @@ impl<'a, 'tcx> Visitor<'tcx> for InteriorVisitor<'a, 'tcx> {
             ty.needs_drop(self.fcx.tcx, self.fcx.param_env)
         };
 
+        let find_parent_expr = |mut hir_id| {
+            let hir = self.fcx.tcx.hir();
+            hir_id = hir.find_parent_node(hir_id)?;
```

Comment:
Hmm, won't this miss the case where the initial ID is already an expression? Not sure how to make a test for that.

Reply:
That's intentional. We want to find the first parent node that is also an expression node. Before we just found the parent node, but now we have an additional check to see if it's an expression and if not we recur.